bin/picca_Pk1D.py

```diff
@@ -70,7 +70,24 @@ def process_all_files(index_file_args):
     # read fits or ascii file
     if args.in_format == 'fits':
         hdul = fitsio.FITS(file)
-        deltas = [Delta.from_fitsio(hdu, pk1d_type=True) for hdu in hdul[1:]]
+        try:
+            deltas = [Delta.from_fitsio(hdu, pk1d_type=True) for hdu in hdul[1:]]
+            running_on_raw_transmission = False
+        except ValueError:
+            print("\nPk1d_type=True didn't work on read in, maybe perfect model? Trying without!")
+            deltas = [delta.from_fitsio(hdu,Pk1D_type=False) for hdu in hdul[1:]]
+            for delta in deltas:
+                delta.ivar=np.ones(delta.delta.shape)*1e10
+                delta.mean_snr=1e5
+                delta.mean_reso=1e-3
+                delta.mean_reso_pix=1e-3
+                delta.exposures_diff = np.zeros(delta.delta.shape)
+                #if args.linear_binning:
+                #    delta.dlambda = np.median(sp.diff(10**d.ll))  #(d.ll[-1]-d.ll[0])/(len(d.ll)-1) #both of those should give the same result, but the first is more explicite, second one should be faster, but this shouldn't be a dominant effect
+                #else:
+                #    delta.dll = np.mean(np.diff(d.ll))  #(d.ll[-1]-d.ll[0])/(len(d.ll)-1) #both of those should give the same result, but the first is more explicite, second one should be faster, but this shouldn't be a dominant effect
+                #delta.mean_resolution_matrix=???
+            running_on_raw_transmission = True
```

Comment:
for the values you used by default, do you think it will have an impact on the result? why not taking larger ones just to be sure?

Reply:
the values are just put there so they can be reported in the output files (and so that intermediate steps trying to select parts of those arrays do not crash), they are used nowhere in the actual calculation.

Reply:
ok makes sense